tests/test_coherence.py

```diff
@@ -0,0 +1,122 @@
+import numpy as np
+from mne_connectivity import make_signals_in_freq_bands
+
+import py_neuromodulation as nm
+
+
+def test_coherence():
+    """Check that coherence features compute properly and match expected values."""
+    # Simulate connectivity data (interaction at specified frequency band)
+    sfreq = 500  # Hz
+    n_epochs = 1
+    n_times = sfreq * 2  # samples
+    fband = (15, 20)  # frequency band of interaction, Hz
+    trans = 2  # transition bandwidth of signal, Hz
+    delay = 50  # samples
+    epochs = make_signals_in_freq_bands(
+        n_seeds=1,
+        n_targets=1,
+        freq_band=fband,
+        n_epochs=n_epochs,
+        n_times=n_times,
+        sfreq=sfreq,
+        trans_bandwidth=trans,
+        connection_delay=delay,
+        ch_names=["seed", "target"],
+        snr=0.7,  # change here requires change in `signal/noise_con` vars below
+        rng_seed=44
+    )
+
+    # Set up py_nm channels info
+    ch_names = epochs.ch_names
+    ch_types = epochs.get_channel_types()
+    channels = nm.utils.set_channels(
+        ch_names=ch_names,
+        ch_types=ch_types,
+        reference="default",
+        bads=None,
+        new_names="default",
+        used_types=tuple(np.unique(ch_types)),
+        target_keywords=None,
+    )
+
+    # Set up pn_nm processing settings
+    settings = nm.NMSettings.get_default()
+    settings.reset()
+    settings.features.coherence = True
+
+    # redefine freq. bands of interest
+    # (accounts for signal-noise transition bandwdith when defining frequencies)
+    settings.frequency_ranges_hz = {
+        "signal": {  # strong connectivity expected
+            "frequency_low_hz": fband[0],
+            "frequency_high_hz": fband[1],
+        },
+        "noise_low": {  # weak connectivity expected from 1 Hz to start of interaction
+            "frequency_low_hz": 1,
+            "frequency_high_hz": fband[0] - trans * 2,
+        },
+        "noise_high": {  # weak connectivity expected from end of interaction to Nyquist
+            "frequency_low_hz": fband[1] + trans * 2,
+            "frequency_high_hz": sfreq // 2 - 1,
+        },
+    }
+    settings.coherence_settings.frequency_bands = ["signal", "noise_low", "noise_high"]
+
+    # only average within each band required
+    settings.coherence_settings.features = {
+        "mean_fband": True, "max_fband": False, "max_allfbands": False
+    }
+
+    # connectivity indices, i.e.: ([0, 1], [1, 0])
+    # FIXME: indices with only 1 channel in seeds/targets raises error with type validatn
+    settings.coherence_settings.channels = [ch_names, ch_names[::-1]]
```

Comment:
One thing I came across was a `ValidationError` from `pydantic` if I tried to pass in indices that contained lists of only one entry, e.g. `[["seed"], ["target"]]`:
```
List should have at least 2 items after validation, not 1 [type=too_short, input_value=['seed'], input_type=list]
```

This means you have to always look at connectivity between more than just 2 channels.

A workaround is to do what I did here and just repeat the indices, e.g. `[["seed", "target"], ["target", "seed"]]`, but that involves unnecessarily computing an extra connection.

I'm not familiar with `pydantic`; is it possible to ignore/whitelist the check for this case?